### PR TITLE
Fix block_event_and_reorg_event_handling test.

### PR DESCRIPTION
**Pull request by Cifko**
## Description
This PR fix 3 issues that were failing:
1) The coinbase_lock_height is 2, so we had to add one empty block to be able to process the UTXO.
2) The TXs were discarded because their input was invalid. Solution was to move it after the processing of the block1.
3) Removal of assert_eq, there is proper one async_assert_eventually.

### CODE REVIEW

**Comment by Cifko** [comment]
Is there a better way to add empty block?

File: base_layer/core/tests/mempool.rs [hunk 1]
```diff
@@ -754,7 +754,7 @@ fn service_request_timeout() {
 #[ignore = "Flaky test that needs to be fixed"]
```
Comment (delta1) [inline base_layer/core/tests/mempool.rs]: If the test is passing reliably now then you can remove this `#[ignore]` attribute so it can run on CI 👍

Comment (Cifko) [reply base_layer/core/tests/mempool.rs]: done. thanks.

File: base_layer/core/tests/mempool.rs [hunk 2]
```diff
@@ -751,10 +751,9 @@ fn service_request_timeout() {
 }
 
 #[test]
-#[ignore = "Flaky test that needs to be fixed"]
 #[allow(clippy::identity_op)]
 fn block_event_and_reorg_event_handling() {
-    // #flaky, this test seems to fail after submiting block B2A to bob.
+    // #flaky, this test seems to fail after submitting block B2A to bob.
```
Comment (delta1) [inline base_layer/core/tests/mempool.rs]: Sorry I forgot to ask to remove this comment as well!

**Review by delta1 — DISMISSED** [review]
Tested locally working, nice one.

File: base_layer/core/tests/mempool.rs (hunk 2 above)
Comment (Cifko) [reply base_layer/core/tests/mempool.rs]: My bad, my spellchecker just noticed that mistake, next time I will read the comments more carefully.

Comment (delta1) [reply base_layer/core/tests/mempool.rs]: No worries 🙃

**Review by delta1 — APPROVED** [review]

**Pull request closed without merging**